### ISSUE TITLE
cluster.rkestate: failed to unmarshal the state file: invalid character '\n' in string escape code

### ISSUE THREAD

**Issue by sbirl** [rancher/rke#2336]
*rke version v1.2.3*

I extracted the rkestate from terraform.tfstate using `jq`.  But `rke etcd snapshot-restore --use-local-state` errors out with **FATA[0000] failed to unmarshal the state file: invalid character '\n' in string escape code**.  I couldnt find any documentation on how rkestate should be formated: JSON? YAML?  Or perhaps there's a better way to obtain the rkestate file during/after the Terraform provisioning process?

I noticed in the rkestate file that values for for certain keys (sshKey, PEMs) are wrapped at the 64th column, followed by a backslash, while other key values go beyond the 64th column.  Could the multi-line wrap be part of the problem?

`jq '.resources[].instances[].attributes.rke_state' /tmp/terraform.tfstate | grep -vw ^null | sed 's/.//;s/.$//' | sed 's/\\"/"/g' | sed 's/\\n/\n/g' >! /tmp/cluster.rkestate`

Thanks.

**Comment by superseb** [comment]
It expects the same file as when `cluster.rkestate` is written by `rke up`, JSON and certificate data splitted using `\n` in the string. In newer version we store the statefile with the snapshot. It should be fairly simple to root cause based on diffing your output and the output from `rke up`. If you think there is a bug here, let me know.

**Comment by sbirl** [comment]
Noted.
Part of the problem is the `sed` used to substitute out the newlines.  Once I modified `sed`, I discovered that the rkestate stored within terraform.tfstate has more data than a proper cluster.rkestate file.

However, my biggest concern for opening up this issue was the lack of documentation/information.  First from the odd error message given by `rke` (*web searching was useless*).  The second issue was not being able to find the format of the rkestate to see if that was reason for the odd error.  So thanks for clarifying that it's JSON.

To further answer my own question about a better way of obtaining the rkestate via Terraform, and for any future readers:
Since the rkestate was being stored in and retrieved from Consul, I had to edit the Terraform file to create `resource "consul_keys" "rkestate"` and then use `terragrunt apply -target=consul_keys.rkestate --auto-approve`.

Finally, long story short, it does not seem to be feasible to use the rkestate extracted from terraform.tfstate for `rke etcd snapshot-restore --use-local-state`.
